core/tx_noncer.go

```diff
@@ -47,30 +55,32 @@ func (txn *txNoncer) get(addr common.InternalAddress) uint64 {
 	// state will mutate db even for read access.
 	txn.lock.Lock()
 	defer txn.lock.Unlock()
-	if _, ok := txn.nonces[addr]; !ok {
-		txn.nonces[addr] = txn.fallback.GetNonce(addr)
+	if !txn.nonces.Contains(addr) {
```

Comment:
I think the ContainsOrAdd library function is a better fit here
https://github.com/hashicorp/golang-lru/blob/v0.6.0/lru.go#L117